src/wm/shim.vala

```diff
@@ -196,60 +194,58 @@ public class ShellShim : GLib.Object
 #if HAVE_MUTTER_5
     private void on_accelerator_activated(uint action, Clutter.InputDevice dev, uint device_id)
 #else
-    private void on_accelerator_activated(uint action, uint device_id)
+    private void on_accelerator_activated(uint action, uint device_id, timestamp)
 #endif
     {
-        HashTable<string,Variant> params = new HashTable<string,Variant>(str_hash, str_equal);
-
-        params.insert("device-id", device_id);
-        params.insert("timestamp", new Variant.uint32(0));
-        params.insert("action-mode", new Variant.uint32(0));
-
-        this.accelerator_activated(action, params);
+        foreach (string accelerator in grabs.get_keys ()) {
+            if (grabs[accelerator] == action) {
+                var params = new GLib.HashTable<string, Variant> (null, null);
+                params.set ("device-id", new Variant.uint32 (device_id));
+#if HAV_MUTTER_5
```

Comment:
It's never going to set this as the timestamp because this should be HAVE_MUTTER_5